File: 0011-container-with-most-water/0011-container-with-most-water.go
func maxArea(height []int) int {
    left, right := 0, len(height) - 1

    maxArea := 0
    for left < right {
        area := min(height[left], height[right]) * (right - left)
        maxArea = max(maxArea, area)

        if height[left] < height[right] {
            left++
        } else {
            right--
        }
    }

    return maxArea
}

File: 0011-container-with-most-water/0011-container-with-most-water.php
class Solution {

    /**
     * @param Integer[] $height
     * @return Integer
     */
    function maxArea($height) {
        $left = 0;
        $right = count($height) - 1;
        $maxArea = 0;

        while ($left < $right) {
            $leftHeight = $height[$left];
            $rightHeight = $height[$right];

            $area = ($right - $left) * min($leftHeight, $rightHeight);
            $maxArea = max($maxArea, $area);

            if ($leftHeight < $rightHeight) {
                $left++;
            } else {
                $right--;
            }
        }

        return $maxArea;
    }
}